### DISCUSSION TITLE
GitHub Actions 在线编译多系统版本

### DISCUSSION THREAD

**Discussion by heiyuan0801** [tw93/Pake#232] [Q&A]
![屏幕截图 2023-01-07 124101](https://user-images.githubusercontent.com/78628929/211131395-172caec6-4f5c-4aa2-a156-309a2a8a596a.png)
在线编译出现这个问题是什么原因

**Comment by tw93** [comment]
这个我看看 不过这里不会影响打包 是代码格式检测

**Reply by heiyuan0801** [reply]
编译完一直错误 然后也不出现文件

**Reply by tw93** [reply]
切到build这里去

**Reply by heiyuan0801** [reply]
![屏幕截图 2023-01-07 133738](https://user-images.githubusercontent.com/78628929/211133014-f75ecd23-674b-465f-9cbf-e2f06a14419c.png)
不行

**Comment by heiyuan0801** [comment]
![屏幕截图 2023-01-07 131122](https://user-images.githubusercontent.com/78628929/211132267-c116dc11-2567-4bdb-997f-8a1aad2ab9e2.png)
![屏幕截图 2023-01-07 131138](https://user-images.githubusercontent.com/78628929/211132270-ef028efc-e56f-4a74-8467-717a259d9afb.png)
一直失败

**Reply by Tlntin** [reply]
从tw93/Pake 同步一下最新代码，即可解决。

**Reply by heiyuan0801** [reply]
![屏幕截图 2023-01-08 134821](https://user-images.githubusercontent.com/78628929/211182445-a791621d-0f8d-42e7-9f98-1f97efc1ccbd.png)
![屏幕截图 2023-01-08 134840](https://user-images.githubusercontent.com/78628929/211182447-9e5dc21f-ceaa-40cc-a637-c434bab8cf00.png)
这里运行完  release 页面怎么什么也没有

**Reply by Tlntin** [reply]
你的tag得大写V开头。

**Reply by miyuesc** [reply]
> 从tw93/Pake 同步一下最新代码，即可解决。

@Tlntin 大佬，请问这是什么问题呀？我按照 actions 的文档执行的，结果打包正常，到上传文件时就出错了，三个版本的脚本都是执行到这一步报错

![image](https://user-images.githubusercontent.com/50617660/211765651-51c1e0bb-993f-4f91-9f10-aa7dde7347a0.png)

**Reply by Tlntin** [reply]
> > 从tw93/Pake 同步一下最新代码，即可解决。
> 
> @Tlntin 大佬，请问这是什么问题呀？我按照 actions 的文档执行的，结果打包正常，到上传文件时就出错了，三个版本的脚本都是执行到这一步报错
> 
> ![image](https://user-images.githubusercontent.com/50617660/211765651-51c1e0bb-993f-4f91-9f10-aa7dde7347a0.png)
放置macos需要的图标（你已完成），`icons/calendar.icns`
```bash
Error failed to bundle project: Failed to build data folders and files: Failed to build data folders and files: Failed to create icon files: Failed to create icon files: path matching png/calendar_256.ico not found.: path matching png/calendar_256.ico not found.
```
运行一下`icns2png.py`脚本，生成对应windows/linux图标。
最后，tag一下新的版本号，然后等待即可。

**Reply by tw93** [reply]
Error failed to bundle project: Failed to create app icon: Failed to create app icon: path matching icons/calendar.icns not found.: path matching icons/calendar.icns not found.
放进去一个 calendar.icns 图标就好了

**Reply by xiyki** [reply]
> @Tlntin 大佬，请问这是什么问题呀？我按照 actions 的文档执行的，结果打包正常，到上传文件时就出错了，三个版本的脚本都是执行到这一步报错

我也遇到了类似的情况，目前还没有找到解决方法
<img width="1427" alt="image" src="https://user-images.githubusercontent.com/83561960/211775935-3549f5c5-2846-4450-9139-4579b0b0a25a.png">

**Reply by Tlntin** [reply]
> > @Tlntin 大佬，请问这是什么问题呀？我按照 actions 的文档执行的，结果打包正常，到上传文件时就出错了，三个版本的脚本都是执行到这一步报错
> 
> 我也遇到了类似的情况，目前还没有找到解决方法 <img alt="image" width="1427" src="https://user-images.githubusercontent.com/83561960/211775935-3549f5c5-2846-4450-9139-4579b0b0a25a.png">

需要设置一下权限，添加GITHUB_TOKEN。
https://stackoverflow.com/questions/70435286/resource-not-accessible-by-integration-on-github-post-repos-owner-repo-ac

**Reply by xiyki** [reply]
> > 我也遇到了类似的情况，目前还没有找到解决方法
> 
> 需要设置一下权限，添加GITHUB_TOKEN。 https://stackoverflow.com/questions/70435286/resource-not-accessible-by-integration-on-github-post-repos-owner-repo-ac

谢谢，在项目的 action 权限设置中将 Workflow permissions 的 Allow GitHub Actions to create and approve pull requests 以及 Read and write permissions 勾选就能正常编译了

**Reply by midpoint** [reply]
我也一直止步在上传文件这一步，百思不得其解，后来才发现是权限的问题。
这是一个坑，GitHub Actions的权限问题，默认不能直接写（例如上传），所以要修改一下权限。

**Reply by Tlntin** [reply]
wiki已更新，增加了actions权限设置说明与截图示例。
